### ISSUE TITLE
Typing in series_test.go

### ISSUE THREAD

**Issue by lyckade** [exercism/xgo#329]
Function name in series_test.go is "Frist"

> // Frist returns the first substring of s with length n.  
> //    Frist(n int, s string) string

I think it should be called First.

**Comment by petertseng** [comment]
I'm trying to understand why this is the case - the comments in https://github.com/exercism/xgo/blob/master/exercises/series/series_test.go#L1 seem to imply the Frist is intentional, but I can't quite grasp why. Wonder if someone with more context can chime in on why this is the case.

**Comment by petertseng** [comment]
My investigation did not seem to turn up the results I wanted - all I got was that this was implemented in #72 but Frist was not discussed there

**Comment by petertseng** [comment]
it's also discussed in #175 but the issue was closed without explanation, again unfortunate

**Comment by kytrinyx** [comment]
I don't recall anything about this. @soniakeys do you have time to jump in here?

**Comment by lyckade** [comment]
After reading #175 and the complete test comments, it is clear why the function is not called First. Because the bonus function should be called First.

But while doing the task I didn't get that point. Also the name "Frist" is not a good name for a function, because it feels like a typo mistake. So maybe it would be better to rename the function or make that point a little bit more clear inside the Readme and the comment. For a non English native speaker, like me, it would be not so confusing. 
But this is just my personal thinking.

**Comment by petertseng** [comment]
I'm trying to figure out what the expected experience is for a student who wishes to work on this exercise:

* First we implement `All` and then an improperly-named function: `func Frist(n int, s string) string`
* Then we try to run `go test -tags asktoomuch`, argh, we can't ever make it pass because the tests include `Frist(6, "01234")` and there's no way to make this pass!
* Then we implement `First(n int, s string) (string, bool)` with the boolean indicating whether the operation was OK, then we can get `go test -tags first` passing.

I guess that's how things should go. So I have some questions along the way.

* I assume at some point we want to show them build constraints? Maybe we should link them to some documentation on that.
* Must the initial function (that doesn't have a bool return) be called `Frist`? Can it be called `UnsafeFirst`, to indicate that it is unsafe if `n > len(s)`? Is it very important that it look like a typo? (I personally don't think so, I personally just think it should be called `UnsafeFirst`, but maybe I miss something!)
* What about the process of initially writing a function that doesn't handle the `n > len(s)` case? Do we want to keep that instead of asking for the correct function from the beginning? From what I can tell, the process teaches that you sometimes have to have an extra return value to indicate some error cases. Do we still want to teach that in this particular exercise?

Basically these questions are aimed at "what would we like students to gain out of doing this exercise?"

Upon review of #72 I saw that it says "we want people to learn out-of-band signaling, starting with Go's `, ok` idiom, so that is the bonus", so the picture is starting to become clearer to me. Does my understanding seem correct?

**Comment by petertseng** [comment]
Since nobody came up with a better idea, I hereby propose to rename `Frist` to `UnsafeFirst` and have attached a PR that does this.

**Comment by kytrinyx** [comment]
+1000 to `UnsafeFirst`. Thank you!
